File: tlow/tlow1.py
def foo(a, b, c, d):
    """Return the value of (a % b) % (c % d)."""

    def modulo(dividend, divisor):
        """Return the remainder of a division attempt."""
        return dividend - dividend // divisor * divisor
  
    return modulo(modulo(a, b), modulo(c, d))

File: tlow/tlow1.java
public class Foo {
  public static void main(String[] args) {}

  private static double foo(double a, double b, double c, double d) {
    return modulo(modulo(a, b), modulo(c, d));
  }
  
  private static double modulo(double dividend, double divisor) {
    return dividend - Math.floor(dividend / divisor) * divisor;
  }
}